src/convex/ConvexAvatarMultiToken.sol

```diff
@@ -0,0 +1,205 @@
+        // NOTE: assume that always will be crv & cvx to convert to stables, while fxs depends on private vaults
+        uint256 totalUsdcEarned;
+        uint256 totalFraxEarned;
+
+        totalUsdcEarned += swapCrvForUsdc(totalCrv);
+        totalUsdcEarned += swapCvxForUsdc(totalCvx);
+        if (totalFxs > 0) {
+            // NOTE: only one hop, avoiding another swap tx form FRAX to USDC
+            totalFraxEarned = swapFxsForUsdc(totalFxs);
+            FRAX.safeTransfer(ownerCached, totalFraxEarned);
+        }
+
+        // 3. Send USDC received to owner
+        USDC.safeTransfer(ownerCached, totalUsdcEarned);
+
+        // Return processed amount
+        processed_ = new TokenAmount[](2);
+        processed_[0] = TokenAmount(address(USDC), totalUsdcEarned);
+        processed_[0] = TokenAmount(address(FRAX), totalFraxEarned);
+
+        emit RewardsToStable(address(USDC), totalUsdcEarned, block.timestamp);
+        emit RewardsToStable(address(FRAX), totalFraxEarned, block.timestamp);
+    }
+
+    function claimAndRegisterRewards() internal returns (uint256 totalCrv_, uint256 totalCvx_, uint256 totalFxs_) {
+        // Update last claimed time
+        lastClaimTimestamp = block.timestamp;
+
+        uint256 length = baseRewardPools.length();
+        for (uint256 i = 0; i < length; i++) {
+            IBaseRewardPool baseRewardPool = IBaseRewardPool(baseRewardPools.at(i));
+            if (baseRewardPool.earned(address(this)) > 0) {
+                baseRewardPool.getReward();
+            }
+        }
+
+        totalCrv_ = CRV.balanceOf(address(this));
+        totalCvx_ = CVX.balanceOf(address(this));
+        totalFxs_ = FXS.balanceOf(address(this));
+
+        if (totalCrv_ == 0) {
+            revert NoRewards();
+        }
+
+        // Emit events for analysis
+        emit RewardClaimed(address(CRV), totalCrv_, block.timestamp);
+        emit RewardClaimed(address(CVX), totalCvx_, block.timestamp);
+        emit RewardClaimed(address(FXS), totalFxs_, block.timestamp);
+    }
+
+    function swapCrvForUsdc(uint256 _crvAmount) internal returns (uint256 usdcEarned_) {
+        uint256 minUsdcExpected = (getCrvAmountInUsdc(_crvAmount) * minOutBpsCrvToUsdc.val) / MAX_BPS;
+        // 1. Swap crv -> eth
+        CRV_ETH_CURVE_POOL.exchange_underlying(
+            1, 0, _crvAmount, (getCrvAmountInEth(_crvAmount) * minOutBpsCrvToUsdc.val) / MAX_BPS
+        );
+
+        // 2. Swap eth -> usdc
+        uint256 ethBal = address(this).balance;
+        IUniswapRouterV3.ExactInputParams memory params = IUniswapRouterV3.ExactInputParams({
+            path: abi.encodePacked(WETH, uint24(500), address(USDC)),
+            recipient: address(this),
+            amountIn: ethBal,
+            amountOutMinimum: minUsdcExpected
+        });
+        usdcEarned_ = UNIV3_ROUTER.exactInput{value: ethBal}(params);
+        // NOTE: conservative approach given https://hackmd.io/@0x534154/Hy6_66gT5
+        UNIV3_ROUTER.refundETH();
```

Comment:
Let's combine the ETH --> USDC swaps of both into a single swap

Reply:
combined in 2d5c0dd5355515d9cc47c3acddc4f7fd02a97486